docs/how-to/hip_porting_guide.rst

```diff
@@ -707,9 +1066,226 @@ architectures with multiple execution units per multiprocessor. For example, the
 AMD GCN architecture has 4 execution units per multiprocessor.
 
 maxregcount
---------------------------------------------------------------------------------
+-----------
 
 Unlike ``nvcc``, ``amdclang++`` does not support the ``--maxregcount`` option.
 Instead, users are encouraged to use the ``__launch_bounds__`` directive since
 the parameters are more intuitive and portable than micro-architecture details
 like registers. The directive allows per-kernel control.
+
+.. note::
+  What does --maxregcount refer to here? Why is it mentioned here as it does
+  not seem to be mentioned anywhere else in this topic. 
```

Comment:
The compiler flag is related to the launch bounds feature on NVIDIA GPUs. The compiler will try to guess the number of registers per thread based on the launch bounds. `-maxregcount X` can be used to override the compiler's decision by enforcing a hard number of registers (X) the compiler must not surpass. If it is unable to make do with `X` registers, it will place additional "registers" into memory instead of hardware registers.